src/cli/init.rs

```diff
@@ -214,8 +218,19 @@ pub async fn execute(args: Args) -> miette::Result<()> {
         let index_url = config.pypi_config.index_url;
         let extra_index_urls = config.pypi_config.extra_index_urls;
 
+        // Dialog with user to create a 'pyproject.toml' or 'pixi.toml' manifest
+        let pyproject  = if !pixi_manifest_path.is_file() && (!args.pixi && !args.pyproject) && pyproject_manifest_path.is_file(){
+            dialoguer::Confirm::new()
+                .with_prompt("\nA 'pyproject.toml' file already exists.\nDo you want to extend it with the [tool.pixi] configuration?")
+                .default(false)
+                .show_default(true)
+                .interact()
+        } else {
+            Ok(args.pyproject)
+        }.into_diagnostic()?;
```

Comment:
This diagnostic only ever applies to dialoguer so I would just move it in the if branch.